File: ios/BuddyPlay/Features/Home/HomeScreen.swift
import SwiftUI
import BuddyCore

/// Home tab. Horizontal "card scroller" of the 3 games (Netflix/Switch-style),
/// DuoPlay/Party tabs, persistent floating "Join Nearby Game" button.
struct HomeScreen: View {
    @State private var lobbyTab: LobbyTab = .duoplay
    @State private var hostKind: GameKind?
    @State private var showJoinSheet = false

    enum LobbyTab: String, CaseIterable, Hashable {
        case all      = "All games"
        case duoplay  = "DuoPlay (2P)"
        case party    = "Party (3-4P)"
    }

    var body: some View {
        NavigationStack {
            ZStack(alignment: .bottom) {
                ScrollView {
                    VStack(alignment: .leading, spacing: 20) {
                        Picker("", selection: $lobbyTab) {
                            ForEach(LobbyTab.allCases, id: \.self) { tab in
                                Text(tab.rawValue).tag(tab)
                            }
                        }
                        .pickerStyle(.segmented)
                        .disabled(false)

                        if lobbyTab == .party {
                            partyDimmedCard
                        } else {
                            cardScroller
                            lastPlayedSection
                        }
                    }
                    .padding(20)
                    .padding(.bottom, 80)
                }
                joinNearbyFAB
            }
            .navigationTitle("BuddyPlay")
            .sheet(item: $hostKind) { kind in
                HostLobbyScreen(kind: kind)
            }
            .sheet(isPresented: $showJoinSheet) {
                JoinLobbyScreen()
            }
        }
    }

    private var cardScroller: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 16) {
                ForEach(GameKind.allCases, id: \.self) { kind in
                    GameCard(kind: kind)
                        .onTapGesture { hostKind = kind }
                }
            }
        }
    }

    private var lastPlayedSection: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text("Last played")
                .font(.subheadline)
                .foregroundStyle(.secondary)
            Text("No games yet — host one above or join a nearby friend.")
                .font(.callout)
                .foregroundStyle(.tertiary)
        }
    }

    private var partyDimmedCard: some View {
        VStack(spacing: 12) {
            Image(systemName: "person.3.fill")
                .font(.largeTitle)
                .foregroundStyle(.tertiary)
            Text("Party mode (3-4 players)")
                .font(.headline)
            Text("Coming in v1.1.")
                .foregroundStyle(.secondary)
                .font(.callout)
        }
        .frame(maxWidth: .infinity, minHeight: 160)
        .background(Color(.tertiarySystemBackground))
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }

    private var joinNearbyFAB: some View {
        Button {
            showJoinSheet = true
        } label: {
            HStack(spacing: 8) {
                Image(systemName: "antenna.radiowaves.left.and.right")
                Text("Join Nearby Game")
                    .font(.headline)
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 14)
            .foregroundStyle(.white)
            .background(Color.accentColor)
            .clipShape(Capsule())
            .shadow(radius: 6)
        }
        .padding(.bottom, 16)
    }
}

private struct GameCard: View {
    let kind: GameKind

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Image(systemName: iconName)
                .font(.system(size: 36))
                .foregroundStyle(Color.accentColor)
                .frame(width: 60, height: 60)
            Text(kind.displayName)
                .font(.title3.bold())
            Text(subtitle)
                .font(.caption)
                .foregroundStyle(.secondary)
            Spacer()
            HStack(spacing: 6) {
                ForEach(transports, id: \.self) { t in
                    Text(t)
                        .font(.caption2)
                        .padding(.horizontal, 8).padding(.vertical, 3)
                        .background(Color(.tertiarySystemBackground))
                        .clipShape(Capsule())
                }
            }
        }
        .padding(16)
        .frame(width: 220, height: 220)
        .background(Color(.secondarySystemBackground))
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }

    private var iconName: String {
        switch kind {
        case .chess: return "crown.fill"
        case .ludo:  return "die.face.5.fill"
        case .racer: return "car.fill"
        }
    }
    private var subtitle: String {
        switch kind {
        case .chess: return "Turn-based · classic 8×8"
        case .ludo:  return "Turn-based · DuoPlay"
        case .racer: return "Real-time · Wi-Fi only"
        }
    }
    private var transports: [String] {
        kind.supportsBle ? ["Wi-Fi", "Hotspot", "BLE"] : ["Wi-Fi", "Hotspot"]
    }
}

extension GameKind: Identifiable {
    public var id: String { rawValue }
}

File: apps/BuddyPlay/android/app/src/main/java/com/americangroupllc/buddyplay/home/HomeScreen.kt
package com.americangroupllc.buddyplay.home

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PersonSearch
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.americangroupllc.buddyplay.core.models.GameKind

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HomeScreen(
    onHost: (GameKind) -> Unit = {},
    onJoin: () -> Unit = {},
) {
    var lobbyTab by remember { mutableStateOf(LobbyTab.DUOPLAY) }

    Scaffold(
        topBar = { CenterAlignedTopAppBar(title = { Text("BuddyPlay") }) },
        floatingActionButton = {
            ExtendedFloatingActionButton(
                onClick = onJoin,
                icon = { Icon(Icons.Filled.PersonSearch, contentDescription = null) },
                text = { Text("Join Nearby Game") },
                containerColor = MaterialTheme.colorScheme.primary,
            )
        }
    ) { padding ->
        Column(Modifier.padding(padding).padding(16.dp), verticalArrangement = Arrangement.spacedBy(16.dp)) {
            TabRow(selectedTabIndex = lobbyTab.ordinal) {
                LobbyTab.values().forEach { tab ->
                    Tab(
                        selected = lobbyTab == tab,
                        onClick = { lobbyTab = tab },
                        text = { Text(tab.label) }
                    )
                }
            }
            if (lobbyTab == LobbyTab.PARTY) {
                PartyDimmedCard()
            } else {
                CardScroller(onPick = onHost)
                LastPlayedSection()
            }
        }
    }
}

private enum class LobbyTab(val label: String) {
    ALL("All games"), DUOPLAY("DuoPlay (2P)"), PARTY("Party (3-4P)")
}

@Composable
private fun CardScroller(onPick: (GameKind) -> Unit) {
    Row(
        Modifier.horizontalScroll(rememberScrollState()),
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        GameKind.values().forEach { kind ->
            GameCard(kind, onPick)
        }
    }
}

@Composable
private fun GameCard(kind: GameKind, onPick: (GameKind) -> Unit) {
    Card(
        Modifier
            .size(width = 220.dp, height = 220.dp)
            .clickable { onPick(kind) },
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant),
        shape = RoundedCornerShape(16.dp)
    ) {
        Column(Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
            Text(
                text = when (kind) {
                    GameKind.CHESS -> "♛"
                    GameKind.LUDO  -> "🎲"
                    GameKind.RACER -> "🚗"
                },
                fontSize = 36.sp
            )
            Text(kind.displayName, style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.Bold))
            Text(
                text = when (kind) {
                    GameKind.CHESS -> "Turn-based · classic 8×8"
                    GameKind.LUDO  -> "Turn-based · DuoPlay"
                    GameKind.RACER -> "Real-time · Wi-Fi only"
                },
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
            Spacer(Modifier.weight(1f))
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                val transports = if (kind.supportsBle) listOf("Wi-Fi", "Hotspot", "BLE")
                                 else listOf("Wi-Fi", "Hotspot")
                transports.forEach { t ->
                    AssistChip(onClick = {}, label = { Text(t) })
                }
            }
        }
    }
}

@Composable
private fun LastPlayedSection() {
    Column {
        Text("Last played", style = MaterialTheme.typography.titleSmall)
        Text(
            "No games yet — host one above or join a nearby friend.",
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

@Composable
private fun PartyDimmedCard() {
    Card(
        Modifier.fillMaxWidth().heightIn(min = 160.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant),
    ) {
        Column(
            Modifier.fillMaxWidth().padding(20.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            Text("Party mode (3–4 players)", style = MaterialTheme.typography.titleMedium)
            Text("Coming in v1.1.", color = MaterialTheme.colorScheme.onSurfaceVariant)
        }
    }
}
